crates/turbopack-ecmascript/src/analyzer/mod.rs

```diff
@@ -2852,6 +2873,97 @@ impl WellKnownObjectKind {
     }
 }
 
+#[derive(Debug, Clone)]
+pub struct RequireContextOptions {
+    pub dir: String,
+    pub include_subdirs: bool,
+    /// this is a regex (pattern, flags)
+    pub filter: Regex,
+}
+
+pub fn regex_from_js(pattern: &str, flags: &str) -> Result<Regex, &'static str> {
+    // rust regex doesn't allow escaped slashes, but they are necessary in js
+    let pattern = pattern.replace("\\/", "/");
+
+    let mut applied_flags = String::new();
+    for flag in flags.chars() {
+        match flag {
+            // case-insensitive: letters match both upper and lower case
+            'i' => applied_flags.push('i'),
+            // multi-line mode: ^ and $ match begin/end of line
+            'm' => applied_flags.push('m'),
+            // allow . to match \n
+            's' => applied_flags.push('s'),
+            // Unicode support (enabled by default)
+            'u' => applied_flags.push('u'),
+            // default in rust, ignore
+            'g' => {}
+            // TODO: maybe error in the future
+            _ => {}
+        }
+    }
+
+    let regex = if !applied_flags.is_empty() {
+        format!("(?{}){}", applied_flags, pattern)
+    } else {
+        pattern
+    };
+
+    Regex::new(&regex).map_err(|_| "could not convert javascript regex to rust regex")
+}
+
+pub fn parse_require_context(args: &Vec<JsValue>) -> Result<RequireContextOptions, &'static str> {
+    if !(1..=3).contains(&args.len()) {
+        return Err("only 1-3 arguments are supported (mode is not supported)");
```

Comment:
Do we have a task to support "mode"? If so, link it here.

Reply:
created one